### PR TITLE
Build: Bundle lib/core-server with ts-up

### PR DESCRIPTION
**Pull request by ndelangen**
related: https://github.com/storybookjs/storybook/issues/18732

### CODE REVIEW

File: code/lib/builder-manager/src/index.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { dirname, join } from 'path';
-import { copy, writeFile, remove } from 'fs-extra';
```
Comment (tmeasday) [inline code/lib/builder-manager/src/index.ts:2]: What's happening with all these changes @ndelangen? Does tsup not like named imports in some cases?

File: code/lib/core-server/src/index.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 // eslint-disable-next-line @typescript-eslint/triple-slash-reference
 /// <reference path="./typings.d.ts" />
 
+//
```
Comment (tmeasday) [inline code/lib/core-server/src/index.ts:4]: ?

File: code/lib/core-server/src/utils/server-statics.ts [hunk 3]
```diff
@@ -3,13 +3,19 @@ import type { Options, StorybookConfig } from '@storybook/core-common';
...
 import { dedent } from 'ts-dedent';
 
-const defaultFavIcon = require.resolve('@storybook/core-server/public/favicon.ico');
+const { pathExists } = fse;
+
+const defaultFavIcon = path.join(
+  path.dirname(require.resolve('@storybook/core-server/package.json')),
+  'public',
+  'favicon.ico'
+);
```
Comment (tmeasday) [inline code/lib/core-server/src/utils/server-statics.ts:18]:
```suggestion
const defaultFavIcon = path.resolve(
require.resolve('@storybook/core-server/package.json'), './public/favicon.ico')
```

**Review by tmeasday — APPROVED** [review]
LGTM

File: code/lib/builder-manager/src/index.ts (hunk 1 above)
Comment (ndelangen) [reply code/lib/builder-manager/src/index.ts:2]: I think what happening is that node now runs in actual ESM mode, or esbuild-register is mimicking that mode more accurately, and `fs-extra` isn';'t actually ESM.

File: code/lib/core-server/src/index.ts (hunk 2 above)
Comment (ndelangen) [reply code/lib/core-server/src/index.ts:4]: I had a commit that was corrupt on CI 😖 I needed to commit something to kick it into action. Will remove.

**Comment by ndelangen** [comment]
Converted into draft, this seems to be a very pivotal package, not easy to convert. I might drop this for now, since I need to work on other stuff.

**Pull request closed without merging**